### DISCUSSION TITLE
Add nano-second support for timestamps for versions prior to .net7

### DISCUSSION THREAD

**Discussion by danielhuntleyalpine** [serilog-contrib/serilog-sinks-grafana-loki#247] [Ideas]
When using this package with versions earlier than `.net 7`, timestamps are published to the closest `ms` rather than the `nanosecond`. This is a request to add support for micro and nano seconds similar to how we handle .net7 and later versions.

I can appreciate that the `Microsecond` and `Nanosecond` functions aren't available in earlier versions but we should still be able to obtain that information from the ticks.

Any help on this would be greatly appreciated! Thanks for taking the time to consider this request.

**Comment by mishamyte** [comment]
Hi @danielhuntleyalpine,

The idea is pretty nice, I like it.

Prob it's a point of small research + impl

**Reply by mishamyte** [reply]
Hey again @danielhuntleyalpine!

Implemented in [v8.3.0](https://www.nuget.org/packages/Serilog.Sinks.Grafana.Loki/8.3.0)

Feel free to use it. The precision is 100ns (which is 1 tick), hope it's enough even for F1-related 🏎️ software 😊

**Reply by danielhuntleyalpine** [reply]
Amazing really appreciate the quick implementation! We will give it a try shortly, thanks for the help!
